pkg/clustermanager/reconcile_logicalcluster.go

```diff
@@ -45,22 +45,51 @@ func (c *controller) reconcileLogicalCluster(key string) error {
 		}
 	} else {
 		c.logger.V(2).Info("reconcile LogicalCluster", "resource", cluster.Name)
-		err := c.processAddLC(cluster)
+		err := c.processAddOrUpdateLC(cluster)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-// TODO: the processAddLC needs to check whether the LogicalCluster has already
-// been added and processed, or whether this is actually an Update.
-func (c *controller) processAddLC(newCluster *lcv1alpha1.LogicalCluster) error {
+// processAddOrUpdateLC: process an added or updated LC object
+func (c *controller) processAddOrUpdateLC(logicalCluster *lcv1alpha1.LogicalCluster) error {
+	switch logicalCluster.Status.Phase {
```

Comment:
you have only 2 if's I suggest just do if instead of case & default
if (LogicalClusterPhaseNotReady && !logicalCluster.Spec.Managed) {
.....
return
}
if  (logicalCluster.Status.Phase =="") {
....
return
}
return nill